src/core/fix-headers.js

```diff
@@ -2,17 +2,25 @@
 // Make sure that all h1-h6 headers (that are first direct children of sections) are actually
 // numbered at the right depth level. This makes it possible to just use any of them (conventionally
 // h2) with the knowledge that the proper depth level will be used
+import { renameElement } from "core/utils";
+
 export const name = "core/fix-headers";
 
-export function run(conf, doc, cb) {
-  const $secs = $("section:not(.introductory)", doc).find(
-    "h1:first, h2:first, h3:first, h4:first, h5:first, h6:first"
-  );
-  $secs.each(function() {
-    let depth = $(this).parents("section").length + 1;
-    if (depth > 6) depth = 6;
-    const h = "h" + depth;
-    if (this.localName.toLowerCase() !== h) $(this).renameElement(h);
-  });
-  cb();
+export function run() {
+  [...document.querySelectorAll("section:not(.introductory)")]
+    .map(sec => sec.querySelector("h1, h2, h3, h4, h5, h6"))
```

Comment:
I recall `:first` being important.

Reply:
`querySelector` will pick first one (either of h1, h2 ...) in every section here, right?

Reply:
Ah, good point 🧐👍